src/protocol/payload_factory.rs

```diff
@@ -18,6 +20,9 @@ impl PayloadFactory {
             Payload::UniCatchupReq(message) => {
                 message.nonce += 1;
             }
+            Payload::MsgDigestSkip(message) => {
+                message.0 = rand::thread_rng().gen::<[u8; 32]>();
```

Comment:
just thinking here, an improvement would be to initialize `rand::thread_rng()` only once and keep it inside the PayloadFactory struct (`self.rnd`), but it could be tricky